spec/features/teachers/user_can_login_spec.rb

```diff
@@ -0,0 +1,62 @@
+require "rails_helper"
+
+describe "As a teacher" do
+  before :each do
+    @teacher = create(:teacher)
+    visit welcome_path
+  end
+
+  it "I can log in successfully" do
+    create(:course,2)
+    create(:student,4)
+    @teacher.courses << Course.first
+    Student.first.courses << Course.first
+    Student.second.courses << Course.first
+    Student.third.course << Course.second
+    Student.fourth.course << Course.second
+
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@teacher)
+
+    expect(page).to eq("Welcome to HereToLearn")
+    fill_in "email", with: @teacher.email
+    fill_in "password", with: @teacher.password
+    click_on "Log In"
+    expect(current_path).to eq(dashboard_path)
+    expect(page).to have_content("Successfully logged in")
+    expect(page).to have_content("Logged in as #{@teacher.first_name} #{teacher.last_name}")
+
+    within "nav" do
```

Comment:
Re: Testing organization style, maybe all the testing of courses/students being present below here should be in a dashboard feature test instead of the log-in test?

Reply:
I think it'll be okay to double up with this. I just wanted there to be a noticeable difference between the welcome page and the dashboard page.